analytics/public-cloud/requests.ts

```diff
@@ -0,0 +1,96 @@
+import { RequestType } from '@prisma/client';
+import prisma from '@/lib/prisma';
+import _isEqual from 'lodash-es/isEqual';
+
+export type CombinedDataPoint = {
+  date: string;
+  'All requests': number;
+  'Edit requests': number;
+  'Create requests': number;
+  'Delete requests': number;
+};
+
+const formatter = new Intl.DateTimeFormat('en', { month: 'short', year: 'numeric' });
+
+function createMonthKey(date: Date) {
+  return formatter.format(date);
+}
+
+export async function combinedRequests() {
+  const requests = await prisma.publicCloudRequest.findMany({
+    select: {
+      created: true,
+    },
+
+    orderBy: {
+      created: 'asc',
+    },
+  });
+
+  const editRequests = await prisma.publicCloudRequest.findMany({
+    where: {
+      type: RequestType.EDIT,
+    },
+    select: {
+      created: true,
+    },
+    orderBy: {
+      created: 'asc',
+    },
+  });
+
+  const createRequests = await prisma.publicCloudRequest.findMany({
+    where: {
+      type: RequestType.CREATE,
+    },
+    select: {
+      created: true,
+    },
+    orderBy: {
+      created: 'asc',
+    },
+  });
+
+  const deleteRequests = await prisma.publicCloudRequest.findMany({
```

Comment:
I believe we can use `Promise.all` to get the performance gain by sending multiple DB calls concurrently.

Reply:
It becomes messy to do it with Promise.all: